pkg/engine/fetchit.go

```diff
@@ -721,63 +687,71 @@ func (hc *FetchitConfig) applyInitial(ctx context.Context, mo *SingleMethodObj,
 }
 
 func (hc *FetchitConfig) getChangesAndRunEngine(ctx context.Context, mo *SingleMethodObj, path string) error {
-	var lc *object.Commit
 	var targetPath string
 	switch mo.Method {
 	case rawMethod:
-		lc = mo.Target.Methods.Raw.lastCommit
 		targetPath = mo.Target.Methods.Raw.TargetPath
 	case kubeMethod:
-		lc = mo.Target.Methods.Kube.lastCommit
 		targetPath = mo.Target.Methods.Kube.TargetPath
 	case ansibleMethod:
-		lc = mo.Target.Methods.Ansible.lastCommit
 		targetPath = mo.Target.Methods.Ansible.TargetPath
 	case fileTransferMethod:
-		lc = mo.Target.Methods.FileTransfer.lastCommit
 		targetPath = mo.Target.Methods.FileTransfer.TargetPath
 	case systemdMethod:
-		lc = mo.Target.Methods.Systemd.lastCommit
 		targetPath = mo.Target.Methods.Systemd.TargetPath
 	default:
 		return fmt.Errorf("unknown method: %s", mo.Method)
 	}
+
 	tp := targetPath
 	if path != "" {
 		tp = path
 	}
-	changesThisMethod, newCommit, err := hc.findDiff(mo.Target, mo.Method, tp, lc)
+
+	working, err := hc.gm.GetCurrentWorkingCommit(mo.Target.Name, mo.Method)
+	if err != nil {
+		return err
+	}
+
+	latest, err := hc.gm.GetLatestCommit(mo.Target.Name)
+	if err != nil {
+		return err
+	}
+
+	changes, err := hc.gm.GetDiff(mo.Target.Name, working, latest)
 	if err != nil {
-		return utils.WrapErr(err, "error method: %s commit: %s", mo.Method, lc.Hash.String())
+		return err
 	}
 
-	hc.setlastCommit(mo.Target, mo.Method, newCommit)
 	hc.update(mo.Target)
```

Comment:
this update should be after you set the workingCommit, below, after this line: https://github.com/containers/fetchit/pull/177/files#diff-f909bb48f0bef398a8f323f6ebb382f1c1b48763963d34b5e90dfe6289b87b97R754